### ISSUE TITLE
Bug: [only-throw-error] only-throw-error.allow[].package unexpectedly works for prefixes

### ISSUE THREAD

**Issue by c-harding** [typescript-eslint/typescript-eslint#11946]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://eslint-online-playground.netlify.app/#eNp1Uk1v2zAM/SuELm6BxMYG7DDvEqAbsNuAobeqB0FmEqeyJEhK08Lwfy/1YTetUR8k4/Hx8VHkyLyTDb6IwSqsg2ct6wdrXIAR/oZg/zhn3H/01miPMMHemQGqndCHsxKukWYYjG6OxKx+cc11ODpzAY2XdfbNLduw4KXR+/5Qn7zRVGycCESveh3qJUJ408Au+K08onyKuoupDve9xrtEXfxkAXIT0Wik0IPPkUILrxa9dL0N24xnz/iSyKQszirdS4UbrmFRKQZ97TD2jbrD7p4k76JJ7DaRO8YDQMUHEgf8Z0NPvbczDmCF8+jWOEWcOaEMLQR3xhmekur77c4Kr9Oq3aqpxmj1uk2D2GIcQNXCw3uZKmNFL35XHgCEUuZCCWN6sxYqK+QTdVJtQIsBCVgNlkKFRNHr1ahgeryqM/cAMKMZiectTYI2oegs2xHDnMUWOWvpbzAdPQBnKY+zDm2cgpY9+kgonXD2aUFz8ref9ff6B2dzzazw/PtLkfKepwRzpkRAH0rxSMjxL4KruXzkJRNcT2x6A9ZJKNQ=

### Repro Code

```TypeScript
import { HttpErrorResponse } from '@angular/common/http';

throw new HttpErrorResponse()
```

### ESLint Config

```javascript
// @ts-check

import { defineConfig } from 'eslint/config';
import tseslint from 'typescript-eslint';

export default defineConfig(
  tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        project: true
      },
    },
    rules: {
      '@typescript-eslint/only-throw-error': [
        'error',
        {
          allow: [{ from: 'package', name: 'HttpErrorResponse', package: '@angular/com' }],
        },
      ],
    },
  },
);
```

### tsconfig

```jsonc
{}
```

### Expected Result

```
Expected an error object to be thrown.
[(@typescript-eslint/only-throw-error)](https://typescript-eslint.io/rules/only-throw-error)
[3:7-30]
```

because `HttpErrorResponse` was only allowed from package `@angular/com`, not from package `@angular/common/http`

### Actual Result

No error: the import seems to be whitelisted.

### Additional Info

I created this issue on @JoshuaKGoldberg’s [suggestion](https://github.com/typescript-eslint/typescript-eslint/issues/11817#issuecomment-3673125244) from https://github.com/typescript-eslint/typescript-eslint/issues/11817.

**Comment by JoshuaKGoldberg** [comment]
Yeah, I think this is a bug coming from an oversight (most likely on my part). I think users should be required to use `*` or some equivalent to indicate glob-style prefix matching.

**Comment by c-harding** [comment]
@JoshuaKGoldberg Alternatively you could only allow prefixes based on full path components: I understand `@angular`, but not `@a`. In any case it should be documented.

**Comment by kirkwaiblinger** [comment]
+1 to the full path component matching as the resolution to this issue (rather than requiring an explicit `**`-type syntax for importing a subpath). 

In other words, 
- `@angular` should match `@angular`, `@angular/foo`, and `@angular/foo/bar`, but not `@angularlol` or `@angularlol/foo`.
- `@angular/foo` should match `@angular/foo` and `@angular/foo/bar` but not `@angular/bar` or `@angular/foobar`

**Comment by amiin-dev** [comment]
I've worked on a fix for this using path-boundary matching. I added tests for the  /  case and made sure the existing  cases still match correctly.

I can open a PR if this approach looks good.

**Comment by amiin-dev** [comment]
I've worked on a fix for this using path-boundary matching. I added tests for the  /  case and made sure the existing  cases still match correctly.

I can open a PR if this approach looks good.

**Comment by amiin-dev** [comment]
I've worked on a fix for this using path-boundary matching. I added tests for the `@angular` / `@angularlol` case and made sure the existing `@types` cases still match correctly.

I can open a PR if this approach looks good.
